EIPS/eip-2566.md

```diff
@@ -0,0 +1,128 @@
+
+Let's compare what we see in the images above with what a user would see if a malicious dapp swaps out the address argument for one of their own. We will leave all other information the same but replace `0x6Aa89e52c9a826496A8f311c1a9db62fd477E256` with `0x9e0A8333984D617562a29a935123F17257aD4dEc`.
+
+<img src="../assets/eip-2566/compromised_tx.png" style="padding-bottom: 20px; padding-top: 20px" width="35%" />
+
+You can see how quiet this slight-of-hand is. The hexadecimal encoded function-execution data has changed to include the new address but it is very unlikely that this non-human readable data field would cause any user to suspect that a dapp is stealing from them. Although this attack would likely be obvious to many examining the dapps front end source code, we cannot assume that all users have the ability to identify such an attack. We should strive to give end users much better security when interacting with contracts. Something like the following can be easily imagined with the `eth_sendTransactionToContractFunction` method.
+
+<img src="../assets/eip-2566/human_readable_tx.png" style="padding-bottom: 20px; padding-top: 20px" width="35%" />
+
+The Ethereum community is trending toward tools that better inform users about their actions. ProviderWallets will remain an integral part of the Ethereum community - serving as the backbone for dapp interconnectivity. The Ethereum RPC API specs should be augmented with methods like `eth_sendTransactionToContract` that allow ProviderWallets to better inform users.
+
+## Specification
+This EIP proposes increasing the set of Ethereum RPC methods to include a new  method - `eth_sendTransactionToContractFunction`. This method roughly parallels `eth_sendTransaction` with the only difference being the human readable `data` field representing the non-encoded arguments to the contract function, and the inclusion of the contract function's `abi` field.
+
+Parameters
+
+1. `Object` - The transaction object
+  * `from`: `DATA`, 20 Bytes - The address the transaction is send from.
+  * `to`: `DATA`, 20 Bytes - (optional when creating new contract) The address the transaction is directed to.
+  * `gas`: `QUANTITY` - (optional, default: 90000) Integer of the gas provided for the transaction execution. It will return unused gas.
+  * `gasPrice`: `QUANTITY` - (optional, default: To-Be-Determined) Integer of the gasPrice used for each paid gas
+  * `value`: `QUANTITY` - (optional) Integer of the value sent with this transaction
+  * `data`: `DATA` - The arguments to the function (not ABI encoded)
```

Comment:
This needs to be defined more clearly.  How are the arguments encoded if not ABI encoded?

Reply:
Good callout - I agree this needs to be more specific. I will update with a slightly different approach. The `data` field remains the same as `eth_sendTransaction` and the only difference between the two functions is the `abi` field. The `data` field allows the EVM to understand what were trying to do. The `abi` field gives us the function name and the argument types. We can parse the argument values from the `data` field and then given that we know the types, implementors can display them in the appropriate formats.